### PR TITLE
Optional yieldIfContendedSafely() on bulkInsert

### PR DESCRIPTION
**Pull request by Electryc**
Allowing to specify an optional query parameter for allowing or not [`yieldIfContendedSafely()`](http://developer.android.com/reference/android/database/sqlite/SQLiteDatabase.html#yieldIfContendedSafely()) during bulk inserts. 

The reasoning behind not wanting the yield is that in case of parallel multithreaded bulkInserts the yield will case ending and starting new transactions and thread context switching in turn causing performance issues. 

In order to keep backwards compatibility and for a more flexible API, we didn't removed the yield completely, but rather let the user choose. By default, the bulkInsert **will** yield. 

_Notice the major increase of performance when not using yield_:

bluk_yield

### CODE REVIEW

**Comment by blundell** [comment]
Add the useage to the demo module

**Comment by Electryc** [comment]
@blundell :+1: 6152aa6

**Comment by blundell** [comment]
I more imagined you would create a new page in the demo that does a yield. Then you can write why you are using yield in that scenario. Not just change the other demo's :-(

File: demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/DatabaseWriter.java [hunk 1]
```diff
@@ -23,10 +24,14 @@ public void bulkSaveDataToFireworksTable(ContentValues[] values) {
     }
 
     private void bulkSaveDataToTable(String table, ContentValues[] values) {
-        Uri uri = createUri(table);
+        Uri uri = createUriWithoutYieldFor(table);
```
Comment (xrigau) [inline demo-extended/src/main/java/com/novoda/sqliteprovider/demo/persistance/DatabaseWriter.java]: wohoo :+1:

**Comment by Electryc** [comment]
All right, I'll provide an example with both yield and without it in order to compare the multithreaded bulkInsert in terms of duration :+1:

**Comment by blundell** [comment]
:+1:  when I said "write what it means" I meant in a textview in the demo, so the peeps using it can read it there n then

**Comment by Electryc** [comment]
@blundell added an example in the sample module where the difference between yield and not yield can be clearly seen.

**Comment by blundell** [comment]
nice i assume the edittexts have hints to know which is which

**Comment by Electryc** [comment]
@blundell yep

File: demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/AddBulkYieldFireworksActivity.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            }
+        }
+        uriSqlFragment.setInfo(shouldYield ? UseCaseFactory.UseCase.BULK_YIELD_ADD : UseCaseFactory.UseCase.BULK_WIHOUT_YIELD);
+    }
+
+    private String createSQL(List<Firework> data) {
+        Firework firework = data.get(0);
+        return TextUtils.replace(
+                DatabaseConstants.RawSql.BULK_INSERT_FIREWORKS,
+                new String[]{"Times", "Na", "Co", "No", "Ft", "Pr", "Sh"},
```
Comment (blundell) [inline demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/AddBulkYieldFireworksActivity.java]: your input data offends me

Comment (Electryc) [inline demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/AddBulkYieldFireworksActivity.java]: This is not really input data, there are just placeholders to be replaced with the computed sql for the users of the demo to see what sql has been executed. Just following the other screens examples :tongue:

Comment (blundell) [inline demo-extended/src/main/java/com/novoda/sqliteprovider/demo/ui/AddBulkYieldFireworksActivity.java]: other demo's are amazing soooo :+1:

**Pull request merged**